### DISCUSSION TITLE
Optional additional app_data on Announces

### DISCUSSION THREAD

**Discussion by SebastianObi** [markqvist/Reticulum#121] [Ideas]
Hi Mark,

I ask here whether the following would possibly make sense to implement. I want to transport a very small amount of additional information for my application in Announces. This data should not be displayed in the normal Announces stream. But still the normal display name like now.

There should be an additional variable fields like in the lxmf messages.

This would have the disadvantage that this change would not be downward compatible with older versions. It is a change to the data structure.

What do you say to this idea?

**Comment by markqvist** [comment]
Can you describe what you are trying to achieve with it, or what problem you are trying to solve? As you mention, it will completely break backwards compatibility, and from the above I don't really see any general benefit that justifies that.

**Reply by SebastianObi** [reply]
I continued to think about my special requirement. You are right. It would be unnecessary to spend the effort only for a special scenario with no other use cases.  I wanted to achieve that my group software can transport something additional with the normal Announces. Which is not displayed in NomadNet/Sideband. Only my own software can read it from these Announces.

In the end, this is also not absolutely necessary. I have now simply defined a 2nd destination with a different type / name. I wanted to save the 2nd Announce.

**Reply by markqvist** [reply]
Makes sense, better to do it like that!

**Reply by SebastianObi** [reply]
Or simply ignore/cut the announce string in sideband/nomadnet after a certain character (newline for example) on the display. It is about not displaying a certain part of the announce string in the user interface. Maybe I try this and make a pull request. Normally newlines wouldn't used at the announced names.

**Reply by markqvist** [reply]
Doing it like that would be a bad design, and I am not going to accept a pull request for that ;)

What is it exactly you are trying to convey in the app data?

**Reply by SebastianObi** [reply]
Of course, as a non-educated programmer, I didn't really think this through. That would be really something very unsightly made. I work then better with the functionality you thought.  You can close this question.
